enterprise_access/apps/bffs/context.py

```diff
@@ -43,23 +54,203 @@ def request(self):
+        staff_enterprise_customer = None
+        has_enterprise_customer_slug_or_uuid = self.enterprise_customer_slug or self.enterprise_customer_uuid
+        if (
+            not enterprise_customer_user_for_requested_customer and
+            has_enterprise_customer_slug_or_uuid and
+            self.user.is_staff
+        ):
+            try:
+                staff_enterprise_customer = self.lms_api_client.get_enterprise_customer_data(
+                    enterprise_customer_uuid=self.enterprise_customer_uuid,
+                    enterprise_customer_slug=self.enterprise_customer_slug,
+                )
+            except Exception as e:  # pylint: disable=broad-except
+                self.add_error(
+                    user_message='Error retrieving enterprise customer data',
+                    developer_message=str(e)
+                )
+
+        # Determine the enterprise customer user to display
+        requested_enterprise_customer = (
+            enterprise_customer_user_for_requested_customer.get('enterprise_customer')
+            if enterprise_customer_user_for_requested_customer else None
+        )
+        enterprise_customer = self._determine_enterprise_customer_for_display(
+            active_enterprise_customer=active_enterprise_customer,
+            requested_enterprise_customer=requested_enterprise_customer,
+            staff_enterprise_customer=staff_enterprise_customer,
+        )
+
+        # Update the context data with the enterprise customer user information
+        self.data = {
+            **self.data,
+            'enterprise_customer': enterprise_customer,
+            'active_enterprise_customer': active_enterprise_customer,
+            'all_linked_enterprise_customer_users': enterprise_customer_users,
+            'staff_enterprise_customer': staff_enterprise_customer,
+        }
+
+    def _determine_enterprise_customer_for_display(
+        self,
+        active_enterprise_customer=None,
+        requested_enterprise_customer=None,
+        staff_enterprise_customer=None,
+    ):
+        """
+        Determine the enterprise customer user for display.
+
+        Returns:
+            The enterprise customer user for display.
+        """
+        if not self.enterprise_customer_slug and not self.enterprise_customer_uuid:
+            # No enterprise customer specified in the request, so return the active enterprise customer user
+            return active_enterprise_customer
+
+        slug_matches_active_enterprise_customer = (
+            active_enterprise_customer and active_enterprise_customer.get('slug') == self.enterprise_customer_slug
+        )
+        uuid_matches_active_enterprise_customer = (
+            active_enterprise_customer and active_enterprise_customer.get('uuid') == self.enterprise_customer_uuid
+        )
+        request_matches_active_enterprise_customer = (
+            slug_matches_active_enterprise_customer or uuid_matches_active_enterprise_customer
+        )
```

Comment:
[nit/non-blocking] could this be extracted into a helper function
Ex. `validate_active_enterprise_customer`